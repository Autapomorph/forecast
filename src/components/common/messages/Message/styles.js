import styled from 'styled-components';

export const StyledMessageWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  padding: 10px;
  border-radius: 5px;
`;

export const StyledMessageContent = styled.span`
  text-align: center;
  font-size: 1.5rem;
`;
