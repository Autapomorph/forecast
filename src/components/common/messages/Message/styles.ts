import styled from 'styled-components/macro';

export const MessageWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  padding: 10px;
  border-radius: 5px;
  font-size: 1.3rem;
`;

export const MessageContent = styled.span`
  text-align: center;
`;
