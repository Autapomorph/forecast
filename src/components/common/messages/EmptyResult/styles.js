import styled from 'styled-components';

import Message from '../Message';

// eslint-disable-next-line import/prefer-default-export
export const StyledEmptyListMessage = styled(Message)``;
