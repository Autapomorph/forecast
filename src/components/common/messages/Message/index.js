import React from 'react';

import { StyledMessageWrapper, StyledMessageContent } from './styles';

const Message = ({ children, ...props }) => (
  <StyledMessageWrapper {...props}>
    <StyledMessageContent>{children}</StyledMessageContent>
  </StyledMessageWrapper>
);

export default Message;
