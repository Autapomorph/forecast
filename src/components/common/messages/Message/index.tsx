import React from 'react';

import * as S from './styles';

const Message: React.FC = ({ children, ...props }): React.ReactElement => (
  <S.MessageWrapper {...props}>
    <S.MessageContent>{children}</S.MessageContent>
  </S.MessageWrapper>
);

export default Message;
