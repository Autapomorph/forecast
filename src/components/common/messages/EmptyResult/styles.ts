import styled from 'styled-components/macro';

import Message from 'components/common/messages/Message';

export const EmptyResultMessage = styled(Message)`
  margin: 20px 0;
`;
