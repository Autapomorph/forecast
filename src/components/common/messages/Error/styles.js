import styled from 'styled-components';

import Message from '../Message';

// eslint-disable-next-line import/prefer-default-export
export const StyledErrorMessage = styled(Message)`
  background: var(--bg-error);
  color: var(--error-text-color);
  font-size: 1.3rem;
`;
