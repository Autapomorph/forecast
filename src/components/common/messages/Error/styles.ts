import styled from 'styled-components/macro';

import Message from 'components/common/messages/Message';

export const ErrorMessage = styled(Message)`
  background: var(--bg-error);
  color: var(--error-text-color);
`;
