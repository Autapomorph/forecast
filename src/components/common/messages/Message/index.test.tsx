import { shallow } from 'enzyme';

import Message from '.';

import { StyledMessageContent } from './styles';

describe('Message tests', () => {
  it('renders properly', () => {
    const wrapper = shallow(<Message />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render children', () => {
    const text = 'text';
    const wrapper = shallow(<Message>{text}</Message>);

    expect(wrapper.find(StyledMessageContent).text()).toEqual(text);
  });
});
