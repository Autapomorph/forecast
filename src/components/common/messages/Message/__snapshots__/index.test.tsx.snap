// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message tests renders properly 1`] = `
<styles__MessageWrapper>
  <styles__MessageContent />
</styles__MessageWrapper>
`;
